hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientDiscoveryServiceBuilder.java

```diff
@@ -162,13 +159,19 @@ private AddressProvider createAddressProvider(ClientConfig clientConfig, Discove
     }
 
     private boolean discoverySpiEnabled(ClientNetworkConfig networkConfig) {
-        return (networkConfig.getDiscoveryConfig() != null && networkConfig.getDiscoveryConfig().isEnabled())
-                || Boolean.parseBoolean(properties.getString(DISCOVERY_SPI_ENABLED));
+        DiscoveryConfig discoveryConfig = networkConfig.getDiscoveryConfig();
+        if (discoveryConfig.getDiscoveryServiceProvider() != null) {
```

Comment:
Is this condition correct?

I always thought that this is related to discovery strategies and they need to be enabled. For example, the following XML configuration:
```xml
        <join>
            <discovery-strategies>
                <node-filter class="com.yourpackage.NodeFilter"/>
                <discovery-strategy class="com.hazelcast.jclouds.JCloudsDiscoveryStrategy" enabled="false">
                </discovery-strategy>
            </discovery-strategies>
        </join>
```
will return `true`, while I think it should return `false`. Isn't that the case?

Reply:
With this configuration `discoveryConfig.getDiscoveryServiceProvider()` will return null. 
Actually I don't know if you can provide a service provider via xml.